### PR TITLE
NativeAOT-LLVM: bring IL backend inline with scanner - use unconstructed type for unbox

### PR DESCRIPTION
**Pull request by yowl**
This PR brings the EEType used for `unbox` in line with the scanner at 

https://github.com/dotnet/runtimelab/blob/20f3949f190f196ab4b958cacbeca1ff76ba3383/src/coreclr/tools/aot/ILCompiler.Compiler/IL/ILImporter.Scanner.cs#L918

We now longer use the `MaximallyConstructableType` type but the `NecessaryTypeSymbol`

Fixes the error at step 8 of https://github.com/dotnet/runtimelab/issues/1992 although attempting to run the produced wasm still fails with a different problem which looks related to the construction of the generic dictionaries

```
E:\tmp\NativeAOTLLVMTest\bin\Debug\net7.0\browser-wasm\native\NativeAOTLLVMTest.js:148
      throw ex;
      ^

RuntimeError: memory access out of bounds
    at __GenericLookupFromType_S_P_CoreLib_System_Collections_Generic_Dictionary_2<System___Canon__System___Canon>_TypeHandle_TKey_System___Canon (<anonymous>:wasm-function[605]:0x9cd6e)
    at S_P_CoreLib_System_Collections_Generic_Dictionary_2<System___Canon__System___Canon>___ctor_2 (<anonymous>:wasm-function[602]:0x9c7ed)
    at S_P_CoreLib_System_Collections_Generic_Dictionary_2<System___Canon__System___Canon>___ctor (<anonymous>:wasm-function[3520]:0x3cdc43)
    at S_P_CoreLib_System_AppContext__SetData (<anonymous>:wasm-function[296]:0x3dd2e)
    at Internal_CompilerGenerated__Module___SetAppContextSwitches (<anonymous>:wasm-function[3103]:0x3b74e6)
    at StartupCodeMain (<anonymous>:wasm-function[182]:0x16ecc)
    at __managed__Main (<anonymous>:wasm-function[3032]:0x3b2ba5)
    at main (<anonymous>:wasm-function[9304]:0x62c6b5)
    at E:\tmp\NativeAOTLLVMTest\bin\Debug\net7.0\browser-wasm\native\NativeAOTLLVMTest.js:983:22
    at callMain (E:\tmp\NativeAOTLLVMTest\bin\Debug\net7.0\browser-wasm\native\NativeAOTLLVMTest.js:7476:15)
    at doRun (E:\tmp\NativeAOTLLVMTest\bin\Debug\net7.0\browser-wasm\native\NativeAOTLLVMTest.js:7530:23)
    at run (E:\tmp\NativeAOTLLVMTest\bin\Debug\net7.0\browser-wasm\native\NativeAOTLLVMTest.js:7545:5)
    at runCaller (E:\tmp\NativeAOTLLVMTest\bin\Debug\net7.0\browser-wasm\native\NativeAOTLLVMTest.js:7453:19)
    at removeRunDependency (E:\tmp\NativeAOTLLVMTest\bin\Debug\net7.0\browser-wasm\native\NativeAOTLLVMTest.js:904:7)
    at receiveInstance (E:\tmp\NativeAOTLLVMTest\bin\Debug\net7.0\browser-wasm\native\NativeAOTLLVMTest.js:1071:5)
    at receiveInstantiationResult (E:\tmp\NativeAOTLLVMTest\bin\Debug\net7.0\browser-wasm\native\NativeAOTLLVMTest.js:1089:5)
```

### CODE REVIEW

**Comment by yowl** [comment]
Cc @SingleAccretion

**Review by SingleAccretion — APPROVED** [review]
Seems we have some infrastructural CI issues.

**Comment by jkotas** [comment]
The error is "Image Build.Windows.10.Amd64.VS2019.Open doesn't exist in pool NetCore-Public"

Could you please update the CI configs to match dotnet/runtime main?

**Comment by yowl** [comment]
Looks like I need to upgrade VS from 2019->2022.

File: src/coreclr/nativeaot/BuildIntegration/Microsoft.NETCore.Native.targets [hunk 1]
```diff
@@ -14,34 +14,33 @@ The .NET Foundation licenses this file to you under the MIT license.
 -->
 <Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
 
-  <PropertyGroup Condition="'$(TargetArchitecture)' == 'wasm'">
-    <NativeCodeGen>llvm</NativeCodeGen>
-  </PropertyGroup>
```
Comment (SingleAccretion) [inline src/coreclr/nativeaot/BuildIntegration/Microsoft.NETCore.Native.targets]: Doesn't this (and below) completely remove the build support for LLVM/WASM?

**Comment by SingleAccretion** [comment]
Looks like we need #1596 with default rooting or some other way to prevent the tests from being trimmed.

File: src/coreclr/nativeaot/BuildIntegration/Microsoft.NETCore.Native.targets (hunk 1 above)
Comment (yowl) [reply src/coreclr/nativeaot/BuildIntegration/Microsoft.NETCore.Native.targets]: oh yeah, thanks, will revert that

**Comment by yowl** [comment]
> Looks like we need #1596 with default rooting or some other way to prevent the tests from being trimmed.



> Looks like we need #1596 with default rooting or some other way to prevent the tests from being trimmed.

Right, will merge up to that.  Looks like its just a week from where we are.

**Review by SingleAccretion — APPROVED** [review]
Latetst changes LGTM.

**Pull request merged**